packages/core/src/events/index.ts

```diff
@@ -246,6 +247,7 @@ export class EventFactory {
       messageId,
     }
 
+    validateEvent(evt)
```

Comment:
Does doing this validation here rather than as a plugin change how errors are surfaced to the user? (I thought there was some validation already happening outside of the plugin in AnalyticsBrowser so the plugin wouldn't actually catch much anyway)